### ISSUE TITLE
Feature Request: Render only on prop change

### ISSUE THREAD

**Issue by Specy** [pixijs/pixi-react#536]
### Description

It would be nice to add an option to disable the default ticker and render the application only whenever props change.

This can either be a deep prop change (so whenever any of the subtree elements of the stage have a prop change) or only at the stage level, this assumes that no children are using context or side effects outside of the react state (context would work too, but we have to assume that the stage is not memoised, in this way even if the children use a context, whenever the value of the context is updated, the stage would re render too).

To recap:
1) (harder, perhaps even degrades performance), detect prop changes in the subtree of the stage so that a render is triggered only when needed
2) Render only when the Stage is rerendered (caused by a react rerender), but add assumption of not using any context in the sub tree, whose value changing doesn't trigger a re render of the stage, and no side effects outside of the react state happen
